src/main/java/bisq/core/dao/blockchain/BsqBlockChain.java

```diff
@@ -599,25 +620,50 @@ public long getProposalFee(int blockHeight) {
         });
     }
 
-    //TODO not impl yet
     @Override
-    public boolean isProposalPeriodValid(int blockHeight) {
-        return lock.read(() -> true);
+    public long getBlindVoteFee(int blockHeight) {
+        return lock.read(() -> getValueAtHeight(blindVoteFees, blockHeight));
     }
 
     @Override
-    public long getBlindVoteFee(int blockHeight) {
+    public long getQuorum(int blockHeight) {
+        return lock.read(() -> getValueAtHeight(quorumByHeight, blockHeight));
+    }
+
+    @Override
+    public long getVoteThreshold(int blockHeight) {
+        return lock.read(() -> getValueAtHeight(voteThresholdByHeight, blockHeight));
+    }
+
+    @Override
+    public long getVoteThreshold() {
+        return lock.read(() -> getVoteThreshold(chainHeadHeight));
```

Comment:
Same here with locks, the lock will be taken in `getVoteThreshold(int blockHeight)` anyway.